### ISSUE TITLE
Question regarding the tca function input

### ISSUE THREAD

**Issue by skatrinli** [cozygene/TCA#13]
I have a question regarding the data matrix (X) input for tca function. I recognize that tca doesn't work if we have low variance across the features (in my case methylation beta values of CpG sites). I excluded the CpG sites with low variance (<10e-5) across samples, but it still doesn't work and I end up removing half of the CpGs. The problem might be that I only have 55 samples. 
However, it is working when I use M-values (logit transformed beta values). My question is: Does using M-values instead of methylation beta values make sense?

**Comment by E-R** [comment]
It is not surprising that a 10e-5 (=1e-4) cutoff for the variance keeps only about half of the features in your data. From my experience, as an example, even in large samples in the order of hundreds, this threshold would remove 30-40% of the CpGs in a 450K array dataset. Since your sample size is in the order of several dozens, you'd expect this threshold to remove even more CpGs than that, as you indicated yourself.

Of course, you can get around the lower bound on the variance in the tca function by applying some transformation such as using M-values (not that even multiplying the data by a large number will do the trick), but this will not really change the fact that very lowly variable features (in their natural levels) are very unlikely to reveal significant associations with any phenotype, especially in small sample sizes (and that's even before considering correction for multiple hypotheses).

The bottom line is that you can technically avoid excluding CpGs based on variance but it is typically not a good idea.

**Comment by skatrinli** [comment]
Thank you so much! It's just a pilot study so I don't really expect to see significant associations with the phenotype. My goal is to get cell-type specific methylation values using the tensor function and evaluate the degree of corelation in my dataset. So, I understand that it's doable while using M-values instead of beta-values.

**Comment by E-R** [comment]
Well, it is technically doable with M-values, but note that (1) once you switch from beta to M-values you lose the statistical interpretation of the model (under a logit transformation the assumption taken by TCA that bulk levels are linear transformations of signals coming from the different cell types becomes unclear and hard to justify) (2) we have not tested the method with M-values. In practice, there may not be very big differences between these two alternatives, but again, I haven't really looked into using TCA with M-values so I can't say much about it.
